src/sql/parse.py

```diff
@@ -259,9 +259,22 @@ def split_args_and_sql(line):
     """
     arg_line, sql_line = line, ""
 
-    # Only separate when json arrow operators used
-    # Otherwise, behavior is unchanged
-    if not any(op in line for op in ["->", "-->"]):
+    # When queries include filenames, they may include SQL keywords
+    #   ex. 'penguins_selected'.csv contains "select"
+    # In these cases, splitting the query leads to parsing errors.
+    # So we ignore any filenames by removing text between double quotes ""
```

Comment:
I understand this now but I'm confused about why this changed was made. the issue reported from the user is about negative numbers, how is that connected to filenames?

Reply:
The user reported that when negative numbers were included in the query, they triggered the same errors that `->` or `-->` did. This is because the parser thinks the `-` character is indicating an argument.

Previously, `split_args_and_sql` would conduct a split only when **both** of these conditions were met:
1. either `->` or `-->` was found in the query
2. any SQL command like `select` was found in the query

This was a bit too specific because it ignored corner cases like negative numbers. I decided to remove the first condition from `split_args_and_sql` so that it would conduct a split _every time_ a SQL command was found in the query, without checking for the arrow operators. So any `%sql` query containing the word `select`, `with`, etc. would now be split.

This change introduced a small bug, where any query containing a word like `select` or `with` in a **filename** would also be split. e.g. `%sqlplot boxplot --table 'file with spaces.csv'`.
This caused errors, because these words weren't acting as SQL commands/operators in this case. That's the reason I included [these lines](https://github.com/ploomber/jupysql/pull/964/files#diff-22cfc6ec3e879941494b36e0dffaf24c100ab50173c413d821ebab94d0627ddfR270-R271). While they aren't directly related to fixing the bug, they catch a specific case where the overall solution is a bit too liberal with its approach.

tldr; to solve this bug and possible future bugs, i made my solution more broad. this happened to be a bit _too_ broad, so I had to include a catch for one specific case (filenames).

I can see why you might feel like this is too confusing or specific. So if you would prefer, I can change it to be more similar to the original solution:
1. either `->`, `-->`, **OR a negative number** was found in the query
2. any SQL command like `select` was found in the query

I just thought that my new solution would be more generalizable in the case that there are even more corner cases that include `-` that we haven't stumbled upon yet. Let me know what you think! @edublancas

Reply:
thanks a lot for the detail explanation. I think the solution makes sense!